### ISSUE TITLE
Change to make track number have common base

### ISSUE THREAD

**Issue by DPH** [SoCo/SoCo#510]
Currently SoCo uses track position that is sometime base 0 and sometimes base 1. This confuses when you retrieve information and then try to use it but need to change it's base. The classic is get_current_track_info() or add_to_queue() which return a base 1 track number, then to use it in play_from queue you need to make it a base  0 track number (subtract 1!). Issue #508 is a perfect example - and started this discussion. 
@ghcs27 notes this is a good idea, but it will be a breaking change however at the moment it is inconsistent.

Below I have documented the scale of the issue as far as I can see it. Please add to this with info you have, and let us have your views. 

SoCo function      | Sonos 	| SoCo           |Notes
------------------- | -----------	| -----------   | ----------------------------------
add_uri_to_queue 	|  base1  |  base1 | although default position is=0!
add_to_queue		|  base1  |  base1 | although default position is=0!
add_multiple_to_queue	| base1  |  base1 |DesiredFirstTrackNumberEnqueued=0
get_current_track_info	| base1  |  base1 |
play_from_queue		| base1  |	base0 |
remove_from_queue	| base1  | base0 |
remove_from_sonos_palylist  | base0 ?  | base0 |
move_in_sonos_laylist |  base0 ?	|  base0 |
reorder_sonos_playlist |   base0 ?	| base0 |

Phonically base 0 would make sense, but it looks like Sonos have largely adopted base 1 !
Thoughts?
Cheers David

**Comment by KennethNielsen** [comment]
I agree it would be nice if it was consistent, but I also think we should stay as close to the protocol as possible. It would be nice to know what Sonos actually does for those last 3 items where there are question marks.

**Comment by DPH** [comment]
Agree  - will try to have a look soon.
Cheers Daivd

**Comment by ghcs27** [comment]
I would definitely vote for consistent base 0, because after all this is a python library, and indices in python are base 0. Any other choice will inevitably irritate users at some point. I know the advantages of staying close to the protocol, but this isn't worth it.

For instance, if you have retrieved the queue as a list `queue` and want to start `queue[i]`, the right call should be `play_from_queue(i)` and not `play_from_queue(i+1)`. The same argument applies to the other functions: A track you have added via `add_to_queue(item, i)` should appear at `queue[i]`, not `queue[i-1]`.

**Comment by DPH** [comment]
Have updated list above to show Sonos have a mix of base 0 and base 1.
If we want to match the Sonos protocol we need to change ```play_from_queue``` and ```remove_from-queue``` to be base 1.
To create our own pythonic standard of base 0 we would need to change 6 functions. 

Obviously any change would not be backward compatible!
Cheers David
